cate/ds/esa_cci_odp.py

```diff
@@ -548,14 +546,10 @@ def open_dataset(self,
                      var_names: VarNamesLike.TYPE = None,
                      protocol: str = None) -> Any:
         time_range = TimeRangeLike.convert(time_range) if time_range else None
-        # TODO (kbernat): support region constraint here
         if region:
-            raise NotImplementedError('EsaCciOdpDataSource.open_dataset() '
-                                      'does not yet support the "region" constraint')
-        # TODO (kbernat): support var_names constraint here
+            region = GeometryLike.convert(region)
```

Comment:
Same here as a few comments above. Not sure if ```GeometryLike``` is fully implemented. But note also, that ```GeometryLike``` should be valid even if a line is provided. Maybe ```PolygonLike``` is better for a region.